hybris/custom/klarnapayment/src/com/klarna/payment/facades/impl/DefaultKPPaymentCheckoutFacade.java

```diff
@@ -598,9 +610,22 @@ public void saveAuthorization(final String authorizationToken, final String paym
 			kpPaymentInfo.setPaymentOption(paymentOption);
 			kpPaymentInfo.setFinalizeRequired(finalizeRequired);
 		}
-
-		kpPaymentInfo.setAuthToken(authorizationToken);
-
+		
+		// Auth Token to be set freshly
+		if(authorizationToken != null && !StringUtils.equals(authorizationToken, kpPaymentInfo.getAuthToken())) {
+			kpPaymentInfo.setAuthToken(authorizationToken);
+		}
+		if(LOG.isDebugEnabled())
+		{
+			if( paymentOption ==  null) {
+				LogHelper.debugLog(LOG,"Auth Callback :: Auth Token for cart"+cart.getCode()+" is :: "+authorizationToken);
+			}
+			else
+			{
+				LogHelper.debugLog(LOG,"Save Auth JS response :: Auth Token for cart"+cart.getCode()+" is :: "+authorizationToken);
```

Comment:
No debug logging in production code

Reply:
Debug logs have been removed
Commit can be found in https://github.com/klarna/SAP-Commerce-klarna-payments/commit/c3d029e9830e600544e03277455cf385023f4a5c